### PR TITLE
File setup

### LINKED ISSUE #2: planning pages
# Setup

- planing pages, readmes documentation

> as a user/developer i want to follow the page repo and production process

- markdown

- this will be developed on `file-setup` branch

  - [ ] write planning documentation in `md`

### LINKED ISSUE #3: intro section
# intro section

> As a user I can easily understand how to use the site

- there is a title to the webpage on `index.html`
- there are labels to the `input` and `button` elements on `index.html`
- there is an instructions `section` on `index.html`
- `code` user messages are clear and succinct on `index.html`

- this will be developed as part of the nav bar on the `file-setup` branch

  - [ ] `html` use `h1` for titles and `p` for paragraph
  - [ ] `css` use `instructions` class to stylize instructions section
  - [ ] `js` use `.innerHTML` to modify `code` message

### LINKED ISSUE #4: input box
# input box

> As a user I can create lists of words

- _there is an input box where words can be added to a list_

- there is an `input` `text` box and add `button` on `index.html`

- this will be developed as part of the nav bar on the `file-setup` branch

  - [ ] `html` use `input` `text` for input box
  - [ ] `css` use `input` class to stylize input section
  - [ ] `js` use `addEventListener` and handler functions to manage input

### LINKED ISSUE #5: remove button
# remove button

> As a user I can edit my list

- there is a remove `button` for the `input` box on `index.html`

- this will be developed on `file-setup` branch

  - [ ] `html` use `input` for for input box
  - [ ] `css` use `input` class to stylize input section
  - [ ] `js` use `addEventListener` and handler functions to manage input buttons

### LINKED ISSUE #6: drop bar
# drop bar

> As a user I can sort my list of words

- there is an `select` and `option` drop bar on `index.html`

- this will be developed on `file-setup` branch

  - [ ] `html` use `select` and `option` for drop bar
  - [ ] `css` use `input` class to stylize input section
  - [ ] `js` use `.target.value` and handler and logic function to manage user options

### LINKED ISSUE #7: message line
# message line

> As a user I can be sure my list only contains words

- there is a `code` section for messages to the user on `index.html`

- this will be developed on `file-setup` branch

  - [ ] `html` use `code` to reference warning section
  - [ ] `css` use `input` class to stylize input section
  - [ ] `js` use `.test` to assure all input characters are letters

### LINKED ISSUE #8: overall stylizing
# overall stylizing

> As a user I want the webpage to be pretty and welcoming

- there is css styling to the different page elements on `index.css`

- this will be developed on `file-setup` branch

  - [ ] `css` use classes to stylize page elements

### LINKED ISSUE #11: footer
# footer

> as a user/developer i can see the code for the page

- this will be developed on `file-setup` branch

- [ ] `html` use `p`, `a` and `svg` to create footer section

- [ ] `css` use `footer` class to stylize footer icon

### PR DESCRIPTION
**Pull request by aviv82**
# word list creator 

## Checklists

- [ ] All CI checks pass
- [ ] The branch works when you pull it and run it locally

### HTML

- [ ] the code is well-formatted
- [ ] the HTML code passes [validator.w3.org](https://validator.w3.org/)
- [ ] there are no inline styles (example: `style='color: red;'`)
- [ ] there are no `<style>` tags with CSS, all styles are hrefs
- [ ] there is no inline JavaScript (example: `onclick='doSomething()'`)
- [ ] there are no `<script>` tags with JS, all JS is in an separate file
- [ ] `id`s are used for JavaScript only, not for CSS
- [ ] semantic tags are used
- [ ] ARIA labels are used where necessary
- [ ] spelling and grammar is correct in all site content

### CSS

- [ ] the code is well-formatted
- [ ] the code does not use any `#` id's
- [ ] styles are responsive

### JavaScript

- [ ] code is clean and easy to read
- [ ] helpful variable names are used
- [ ] all comments are clear and help to understand the code
- [ ] there is no unused code in comments
- [ ] all file names are helpful and match their exports

#### src/data.js

- [ ] the file does not import anything
- [ ] the file only exports JS data (primitives, arrays, objects, ...)
- [ ] the file does not contain any functions or logic
- (optional) the data is documented with JSDocs

#### src/init

- [ ] there is an `index.js` that is included by the `index.html` file
- [ ] `../listener` files are imported
- [ ] there is no code that needs to run _after_ the page is initialized
- [ ] any other files have helpful names

#### src/listeners

- [ ] the DOM can be queried to find elements
- [ ] the DOM _is not_ be modified
- [ ] handlers are imported and used as callbacks to event listeners
- [ ] no other functions are imported or used
- [ ] the file name makes sense for the listener

#### src/handlers

- [ ] handlers have a JSDoc comment
- [ ] the function name matches the file name
- [ ] handlers are used as a callback to `.addEventListener` somewhere in the program
- [ ] functions from `/logic` or `/procedures` files may be imported and called
- [ ] functions from `/handlers` may be imported and attached to the DOM with event listeners
- [ ] data from `/data.js` may be imported and used
- [ ] handlers may read and write to the DOM
- [ ] handlers _do not_ return values that you will need later in the program

#### src/components

- [ ] A DOM element is returned
- [ ] handlers can be imported and used as callbacks to event listeners
- [ ] _logic_, _components_ and _procedures_ can also be imported
- [ ] there is an HTML test file to render the component with different inputs
- [ ] each component has a `.test.html` file
- [ ] bonus points for a unit test `.spec.js` file

#### src/logic

- [ ] the handler has a complete and correct JSDoc comment
- [ ] the function name matches the file name
- [ ] other functions from `/logic` may be imported and called
- [ ] the function _does not_ read or write to the DOM
- [ ] the function _does not_ log to the console
- [ ] each logic function has a `.spec.js` file
  - [ ] tests are simple with no extra code
  - [ ] tests are well-named
  - [ ] tests cover common use cases (bonus for edge cases)
  - [ ] tests check for side-effects (if necessary)

### CODE REVIEW

File: client/src/handlers/sort-words.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ export const sortWords = (event) => {
     sort the words in data.words using the logic function
     assign the newly sorted list to a variable named `sorted`
   */
-
+  // data.sort = howToSort; < -- unnecessary
```
Comment (ThibautHumblet) [inline client/src/handlers/sort-words.js]: If code is unnecessary and commented, you better remove it 😉

File: client/src/logic/sort-strings.js [hunk 2]
```diff
@@ -14,6 +14,100 @@
...
+  if (sortType === 'shortest') {
+    return tempArray.sort(sortLength);
+  }
+  if (sortType === 'longest') {
+    return tempArray.sort(sortLength).reverse();
+  }
+  return tempArray;
+};
+
+// case sensitive reference
```
Comment (ThibautHumblet) [inline client/src/logic/sort-strings.js]: Great to reference your sources. But in my opinion, it's better to do this in the ReadMe.md to reduce clutter in your codebase. 😄

**Review by ThibautHumblet — CHANGES_REQUESTED** [review]
Excellent job @aviv82 !

Some things I liked while reviewing your code:

- The use of your multiple readme files to explain your imports/exports 📄 
- You provided some general info what your project's about ℹ️ 
- You referenced your sources (but I've also added some feedback to do this in your readme) 📎 
- You've written some tests 🧪 
- You used the principles of SoC (seperation of concerns) 🖊️ 

I've left some minor feedback but in general: great work! 🔥 

Let me know when you implemented the feedback.
Cheers!

**Review by ThibautHumblet — APPROVED** [review]

**Review by ThibautHumblet — CHANGES_REQUESTED** [review]
(Did an oopsie by approving all the code when I meant to approve just your last changes) 😂

File: client/src/logic/sort-strings.js (hunk 2 above)
Comment (aviv82) [reply client/src/logic/sort-strings.js]: Thank you Thibaut. Edited as you suggested! 🙂

**Review by ThibautHumblet — APPROVED** [review]
Looks great!

Well done @aviv82 🥇

**Comment by gelilaa** [comment]
Great to see what you have done @aviv82 🚀 .  But it is a lot of work in one branch. Next time try to create different branches for your features.(e.g.: logic-branch, CSS-html-branch, planning-branch, src-branch....) 👍🏾

**Review by gelilaa — APPROVED** [review]

**Comment by aviv82** [comment]
> Great to see what you have done @aviv82 🚀 . But it is a lot of work in one branch. Next time try to create different branches for your features.(e.g.: logic-branch, CSS-html-branch, planning-branch, src-branch....) 👍🏾

thank you Gelila! will do. 😀
